### PR TITLE
KAFKA-4200: Fix throttle argument in kafka-reassign-partitions.sh

### PR DESCRIPTION
**Pull request by benstopford**
Simple jira which alters two things: 
1. kafka-reassign-partitions --verify prints Throttle was removed regardless of whether a throttle was applied. It should only print this if the value was actually changed.
2. --verify should exception if the —throttle argument. (check generate too)

To test this I extracted all validation logic into a separate method and added a test which covers the majority of combinations. The validation logic was retained as is, other than implementing (2) and adding validation to the --broker-list option which you can currently apply to any of hte main actions (where it is ignored). Requirement 1 was tested manually (as it's just println). 

Testing:
- Build passes locally. 
- System test reassign_partitions_test.py also passes.

### CODE REVIEW

**Comment by benstopford** [comment]
Build failure was a timeout in KStreamAggregationIntegrationTest which I don't think is related. Rerunning

File: core/src/main/scala/kafka/admin/ReassignPartitionsCommand.scala [hunk 1]
```diff
@@ -88,29 +78,34 @@ object ReassignPartitionsCommand extends Logging {
   }
 
   def removeThrottle(zkUtils: ZkUtils, partitionsToBeReassigned: Map[TopicAndPartition, scala.Seq[Int]], reassignedPartitionsStatus: Map[TopicAndPartition, ReassignmentStatus]): Unit = {
```
Comment (junrao) [inline core/src/main/scala/kafka/admin/ReassignPartitionsCommand.scala]: Could we make this private?

File: core/src/test/scala/unit/kafka/admin/ReassignPartitionsCommandArgsTest.scala [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+    val args = Array(" ")
+    shouldFailWith("Command must include exactly one action: --generate, --execute or --verify", args)
+  }
+
+  /**
+    * UNHAPPY PATH: EXECUTE ACTION
+    */
+
+  @Test
+  def shouldAllowThrottleOptionOnExecute(): Unit = {
```
Comment (junrao) [inline core/src/test/scala/unit/kafka/admin/ReassignPartitionsCommandArgsTest.scala]: This seems a happy path unlike the comment above.

**Comment by junrao** [comment]
Thanks for the patch. LGTM. Just a couple of minor comments.

File: core/src/main/scala/kafka/admin/ReassignPartitionsCommand.scala [hunk 3]
```diff
@@ -30,31 +29,24 @@ import org.apache.kafka.common.security.JaasUtils
...
         verifyAssignment(zkUtils, opts)
-      else if(opts.options.has(opts.generateOpt))
+      }
+      else if(opts.options.has(opts.generateOpt)) {
         generateAssignment(zkUtils, opts)
-      else if (opts.options.has(opts.executeOpt))
+      }
+      else if (opts.options.has(opts.executeOpt)){
         executeAssignment(zkUtils, opts)
+      }
```
Comment (ijuma) [inline core/src/main/scala/kafka/admin/ReassignPartitionsCommand.scala]: Why were the braces added here?

**Comment by ijuma** [comment]
Thanks for the PR. I'll merge it as Jun has reviewed it and you have addressed the minor comments.

**Comment by benstopford** [comment]
Comments all addressed.

**Pull request closed without merging**